src/NuGet.Core/NuGet.ProjectManagement/NuGetProject.cs

```diff
@@ -35,6 +37,14 @@ protected NuGetProject(Dictionary<string, object> metadata)
             get { return InternalMetadata; }
         }
 
+        public string Guid
```

Comment:
Idea here is to map a start batch event with corresponding end batch event, imagine if we support parallel operations on multiple projects in future then we mayneed to map these start/end events. It can't be achieved by generating based on operation.